chain/vm/vm.go

```diff
@@ -292,12 +292,12 @@ func (vm *VM) ApplyMessage(ctx context.Context, cmsg types.ChainMsg) (*ApplyRet,
 	}
 
 	pl := PricelistByEpoch(vm.blockHeight)
+
 	msgGasCost := pl.OnChainMessage(cmsg.ChainLength())
-	// TODO: charge miner??
 	if msgGasCost > msg.GasLimit {
 		return &ApplyRet{
 			MessageReceipt: types.MessageReceipt{
-				ExitCode: exitcode.SysErrOutOfGas,
+				ExitCode: exitcode.SysErrSenderStateInvalid,
```

Comment:
This appears to be incorrect. Tests are failing at the moment because the chain validation suites are expecting `SysErrOutOfGas`.

Reply:
I think this was accidentally added in the last change to address a comment.

Reply:
Hmm, it is edge case. If validation is failing here let's go with whatever validation is saying.